src/custom-elements/src/components/spinner/spinner.tsx

```diff
@@ -0,0 +1,51 @@
+import { Component, Prop, Watch } from '@stencil/core';
+import { TEXT_COLORS } from '../../models/color';
+import { ICON_SIZES } from '../../models/size';
+
+@Component({
+  tag: 'chi-spinner',
+  styleUrl: 'spinner.scss',
+  scoped: true
+})
+export class Spinner {
+  /**
+   *  to set size of a spinner { xs, sm, sm--2, sm--3, md, lg, xl, xxl }.
+   */
+  @Prop({ reflectToAttr: true }) size: string;
+
+  /**
+   *  to set color of a spinner { primary, success, warning, danger, muted }.
+   */
+  @Prop({ reflectToAttr: true }) color: string;
+
```

Comment:
Please, add backdrop as an attribute that should render a div.a-backdrop inside the custom element if set to true. Backdrop is a pseudo component that will be used here in spinner, in the modal and in the drawer. But there are no plans to create a custom element itself.